backend/src/nodes/impl/upscale/tile_blending.py

```diff
@@ -0,0 +1,214 @@
+                    # we can't use all the overlap of the current tile, so we have to cut it off
+                    diff = o.start - self.last_end_overlap
+                    tile = tile[:, diff:, ...]
+                    h, w, c = get_h_w_c(tile)
+                    o = TileOverlap(self.last_end_overlap, o.end)
+
+                # copy over the part that doesn't need blending (yet)
+                self.result[
+                    :, self.offset + o.start : self.offset + w - o.start, ...
+                ] = tile[:, o.start * 2 :, ...]
+
+                # blend the overlapping part
+                blend_size = o.start * 2
+                blend = self._get_blend(blend_size)
+
+                left = self.result[
+                    :, self.offset - o.start : self.offset + o.start, ...
+                ]
+                right = tile[:, :blend_size, ...]
+
+                self.result[
+                    :, self.offset - o.start : self.offset + o.start, ...
+                ] = _fast_mix(left, right, blend)
+
+                self.offset += w - o.total
+                self.last_end_overlap = o.end
+        else:
+            assert w == self.width
+            assert h > o.total
+
+            if self.offset == 0:
+                # the first tile is copied in as is
+                self.result[:h, :, ...] = tile
+
+                assert o.start == 0
+                self.offset += h - o.end
+                self.last_end_overlap = o.end
+
+            else:
+                assert self.offset < self.height, "All tiles were filled in already"
+
+                if self.last_end_overlap < o.start:
+                    # we can't use all the overlap of the current tile, so we have to cut it off
+                    diff = o.start - self.last_end_overlap
+                    tile = tile[diff:, :, ...]
+                    h, w, c = get_h_w_c(tile)
+                    o = TileOverlap(self.last_end_overlap, o.end)
+
+                # copy over the part that doesn't need blending
+                self.result[
+                    self.offset + o.start : self.offset + h - o.start, :, ...
+                ] = tile[o.start * 2 :, :, ...]
+
+                # blend the overlapping part
+                blend_size = o.start * 2
+                blend = self._get_blend(blend_size)
+
+                left = self.result[
+                    self.offset - o.start : self.offset + o.start, :, ...
+                ]
+                right = tile[: o.start * 2, :, ...]
+
+                self.result[
+                    self.offset - o.start : self.offset + o.start, :, ...
+                ] = _fast_mix(left, right, blend)
+
+                self.offset += h - o.total
+                self.last_end_overlap = o.end
+
+    def get_result(self) -> np.ndarray:
+        # if self.direction == BlendDirection.X:
+        #     assert self.offset == self.width
+        # else:
+        #     assert self.offset == self.height
```

Comment:
🤔

Reply:
Turns out, assertions that ensure correctness can get in the way when you're trying to see the incorrect result to understand what is wrong. Well, it works now, so I just forgot to uncomment them.